### ISSUE TITLE
Replace CONDENSE with string function: Syntax error when used with a structure

### ISSUE THREAD

**Issue by ConjuringCoffee** [SAP/abap-cleaner#371]
Hi Jörg-Michael, the rule "Replace CONDENSE with string function" causes a syntax error if `CONDENSE ` was used on a structure.

Example:

```abap
TYPES: BEGIN OF ly_structure,
         field TYPE c LENGTH 10,
       END OF ly_structure.

DATA ls_structure TYPE ly_structure.

CONDENSE ls_structure.
```

The last line is converted to:

```abap
ls_structure = condense( ls_structure ).
```

This causes the following syntax error:

> "LS_STRUCTURE" must be a character-like data object (data type C, N, D, T, or STRING).

This error is very similar to: https://github.com/SAP/abap-cleaner/issues/318

**Comment by jmgrassau** [comment]
Hi ConjuringCoffee, 

thanks for reporting this! I now added a similar option to "Replace CONDENSE with string function" as already done for "Replace TRANSLATE with string functions":

![image](https://github.com/user-attachments/assets/e76aa7ac-019d-4fa2-9470-1561ca4f6b88)

With this change, the first CONDENSE will never be replaced (because it is sure to be structured), while the second CONDENSE will only be replaced if the new option is deactivated (but it is active by default). 

Kind regards,
Jörg-Michael

**Comment by jmgrassau** [comment]
Hi ConjuringCoffee, 

thanks again for reporting this issue – the new option is now available in version 1.20.1, which was just released!

Kind regards,
Jörg-Michael

**Comment by ConjuringCoffee** [comment]
Great, thank you!
